extension/src/popups/MinervaMenu.js

```diff
@@ -21,21 +23,27 @@ function displayClassRankings() {
     });
 }
 
-window.onload = function() {
-    chrome.storage.local.get(["student_name"], (result) => {
-        if (result.hasOwnProperty("student_name")) {
-            transitionToMainMenu(result["student_name"]);
+function diplayCurrentStudentInfo(studentName) {
+    chrome.runtime.sendMessage({
+        type: "FetchCurrentStudentInfo",
+        studentName: studentName
+    }, (response) => {
+        if (response.status) {
+            renderCurrentStudentInfo(response);
+            filterAvailableStoreStickers(response.owned_badges);
         } else {
-            chrome.tabs.create({url: chrome.extension.getURL("StudentRegistration.html")});
+            let studentInfoBody = document.getElementById("card-body-youPanel");
+            studentInfoBody.innerHTML = "No information could be found";
         }
     });
+}
+
+window.onload = function() {
+    let studentHandle = getStudentHandle();
```

Comment:
Use `async/await` here
I am not too sure this file shares the same storage over the lifecycle and I am hesistant to make `ApiInteface` shared between background and this file. The reason is that there will be two copies of ApiInterface at runtime due to how webpack glue things. Let's push this to the background.js

```
window.onload = async() => {
    // eslint .... disable ... ...
    let result = await new Promise((resolve, reject) => {
        chrome.runtime.sendMessage({
            type: "GetStudentName"
        }, (response) => {
            resolve(response);
        });
    });

    if (result.statuts) {
        renderRanking...
        renderStudentProfile(result.name)
    ) else {
        open registration page
    }
});
```

Reply:
refactored.